### PR TITLE
feat: Add DaCo String DML statements support #1126

### PR DESCRIPTION
**Pull request by sachin-shitole-paccar**

### CODE REVIEW

File: server/src/test/java/org/eclipse/lsp/cobol/usecases/TestDAFStringReplaceStatement.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.eclipse.lsp.cobol.service.delegates.validations.SourceInfoLevels;
+import org.eclipse.lsp.cobol.usecases.engine.UseCaseEngine;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.junit.jupiter.api.Test;
+
+/** Tests the DAF STRING FILL statement */
```
Comment (temanbrcom) [inline server/src/test/java/org/eclipse/lsp/cobol/usecases/TestDAFStringReplaceStatement.java:24]: "STRING REPLACE"

**Review by grianbrcom — APPROVED** [review]
:pick:

File: server/src/test/java/org/eclipse/lsp/cobol/usecases/TestDAFStringAddStatement.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            "1",
+            new Diagnostic(
+                null,
+                "Variable GBR4 is not defined",
+                DiagnosticSeverity.Error,
+                SourceInfoLevels.ERROR.getText()),
+            "2",
+            new Diagnostic(
+                null,
+                "Syntax error on '.' expected {ALL, DFHRESP, DFHVALUE, FALSE, HIGH-VALUE, HIGH-VALUES, LOW-VALUE, LOW-VALUES, NULL, NULLS, QUOTES, SPACE, SPACES, TRUE, ZERO, ZEROES, ZEROS, '01-49', '66', '77', '88', INTEGERLITERAL, NUMERICLITERAL, NONNUMERICLITERAL, IDENTIFIER, FINALCHARSTRING}",
```
[the remark this answers is not in the document]
Comment (sachin-shitole-paccar) [reply server/src/test/java/org/eclipse/lsp/cobol/usecases/TestDAFStringAddStatement.java:66]: "Syntax error on '.' expected {ALL, DFHRESP, DFHVALUE, FALSE, HIGH-VALUE, HIGH-VALUES, LOW-VALUE, LOW-VALUES, NULL, NULLS, QUOTES, SPACE, SPACES, TRUE, ZERO, ZEROES, ZEROS, '01-49', '66', '77', '88', INTEGERLITERAL, NUMERICLITERAL, NONNUMERICLITERAL, IDENTIFIER, FINALCHARSTRING}"
Do you mean this message?

[the remark this answers is not in the document]
Comment (sachin-shitole-paccar) [reply server/src/test/java/org/eclipse/lsp/cobol/usecases/TestDAFStringAddStatement.java:66]: @temanbrcom Could you please confirm my understanding?

[the remark this answers is not in the document]
Comment (temanbrcom) [reply server/src/test/java/org/eclipse/lsp/cobol/usecases/TestDAFStringAddStatement.java:66]: Exactly, you may see the `TestExecCicsShowsErrorCorrectly` as an example. IDEA will split a string literal automatically if you hit enter inside it.

[the remark this answers is not in the document]
Comment (sachin-shitole-paccar) [reply server/src/test/java/org/eclipse/lsp/cobol/usecases/TestDAFStringAddStatement.java]: Thanks. Done.

File: server/src/test/java/org/eclipse/lsp/cobol/usecases/TestDAFStringDeleteStatement.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.eclipse.lsp.cobol.service.delegates.validations.SourceInfoLevels;
+import org.eclipse.lsp.cobol.usecases.engine.UseCaseEngine;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.junit.jupiter.api.Test;
+
+/** Tests the DAF STRING FILL statement */
```
[the remark this answers is not in the document]
Comment (sachin-shitole-paccar) [reply server/src/test/java/org/eclipse/lsp/cobol/usecases/TestDAFStringDeleteStatement.java]: Yes correct. Done.

File: server/src/test/java/org/eclipse/lsp/cobol/usecases/TestDAFStringAddStatement.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                null,
+                "Variable GBR4 is not defined",
+                DiagnosticSeverity.Error,
+                SourceInfoLevels.ERROR.getText()),
+            "2",
+            new Diagnostic(
+                null,
+                "Syntax error on '.' expected {ALL, DFHRESP, DFHVALUE, FALSE, HIGH-VALUE, HIGH-VALUES, LOW-VALUE" +
+                        ", LOW-VALUES, NULL, NULLS, QUOTES, SPACE, SPACES, TRUE, ZERO, ZEROES, ZEROS, '01-49', '66', '77'" +
+                        ", '88', INTEGERLITERAL, NUMERICLITERAL, NONNUMERICLITERAL, IDENTIFIER, FINALCHARSTRING}",
```
Comment (grianbrcom) [inline server/src/test/java/org/eclipse/lsp/cobol/usecases/TestDAFStringAddStatement.java:68]: Our style checker expects that lines will start with the `+` sign.

**Comment by sachin-shitole-paccar** [comment]
@temanbrcom Could you please complete the review and merge the PR?

**Review by temanbrcom — APPROVED** [review]
🦘

**Pull request merged**